src/shogun/kernel/PolyKernel.cpp

```diff
@@ -71,8 +71,17 @@ void CPolyKernel::init()
 	m_gamma = 1.0;
 
 	set_normalizer(new CSqrtDiagKernelNormalizer());
-	SG_ADD(&degree, "degree", "Degree of polynomial kernel", ParameterProperties::HYPER);
-	SG_ADD(&m_c, "c", "The kernel is inhomogeneous if the value is higher than 0", ParameterProperties::HYPER);
-	SG_ADD(&m_gamma, "gamma", "Scaler for the dot product", ParameterProperties::HYPER);
+	SG_ADD(
+	    &degree, "degree", "Degree of polynomial kernel",
+	    ParameterProperties::HYPER);
+	SG_ADD(
+	    &m_c, "c", "The kernel is inhomogeneous if the value is higher than 0",
+	    ParameterProperties::HYPER);
+	SG_ADD(
+	    &m_gamma, "gamma", "Scaler for the dot product",
+	    ParameterProperties::HYPER | ParameterProperties::AUTO, [this]() {
+		    return make_any(
```

Comment:
I like this approach but there is a tricky problem.

How do we ensure it already has the features initialized?

Reply:
@lisitsyn i think adding here a REQUIRE() shoudl be enough or?